### PR TITLE
Clean up no undefined variables

### PR DESCRIPTION
**Pull request by jhgg**

### CODE REVIEW

**Comment by jhgg** [comment]
Ugh. I branched from wrong... one sec.

**Comment by jhgg** [comment]
shut up coveralls. -0.001 is rounding error ¬_¬

**Comment by woodb** [comment]
> shut up coveralls. -0.001 is rounding error ¬_¬

It's actually not even possible, there are less than 3,000 LoC in the repository, that's like 0.03 lines of code (??)

File: graphql/core/validation/rules.py [hunk 1]
```diff
@@ -328,22 +329,22 @@ def undefined_var_by_op_message(var_name, op_name):
...
     def enter_VariableDefinition(self, node, *args):
-        self.defined_variable_names[node.variable.name.value] = True
+        self.defined_variable_names.add(node.variable.name.value)
 
     def enter_Variable(self, variable, key, parent, path, ancestors):
         var_name = variable.name.value
         if var_name not in self.defined_variable_names:
-            is_fragment = lambda node: isinstance(node, ast.FragmentDefinition)
-            within_fragment = any(is_fragment(node) for node in ancestors)
+            within_fragment = any(isinstance(node, ast.FragmentDefinition) for node in ancestors)
```
Comment (rawls238) [inline graphql/core/validation/rules.py]: nice

**Comment by rawls238** [comment]
lgtm

**Comment by dittos** [comment]
lol coveralls

**Pull request merged**